m2cgen/ast/interpreters/code_generator.py

```diff
@@ -50,13 +48,11 @@ def decrease_indent(self):
         assert self._current_indent >= 0, (
             "Invalid indentation: {}".format(self._current_indent))
 
-    def add_code_line(self, line, newline=True):
-        indent = "".join([" "] * self._current_indent)
-        self.code += indent + line
-        if newline:
-            self.code += "\n"
+    def add_code_line(self, line):
+        indent = "" * self._current_indent
```

Comment:
I believe this update breaks the indentation logic.

Reply:
Good catch! Fixed.